plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +84,222 @@ protected DistributedQueryRunner createQueryRunner()
-        // for assertions.
-
-        String metadataFileSuffix = "metadata.json";
-        assertEquals(counts.forPathContaining(metadataFileSuffix).forOperation(INPUT_FILE_NEW_STREAM).sum(), 1);
-        // getLength is cached, so only assert number of different InputFile instances for a file
-        assertEquals(counts.forPathContaining(metadataFileSuffix).forOperation(INPUT_FILE_GET_LENGTH).get().size(), 0);
-        assertEquals(counts.forPathContaining(metadataFileSuffix).forOperation(INPUT_FILE_EXISTS).sum(), 0);
-
-        String snapshotFilePrefix = "/snap-";
-        assertEquals(counts.forPathContaining(snapshotFilePrefix).forOperation(INPUT_FILE_NEW_STREAM).sum(), 1);
-        // getLength is cached, so only assert number of different InputFile instances for a file
-        assertEquals(counts.forPathContaining(snapshotFilePrefix).forOperation(INPUT_FILE_GET_LENGTH).get().size(), 1);
-        assertEquals(counts.forPathContaining(snapshotFilePrefix).forOperation(INPUT_FILE_EXISTS).sum(), 0);
-
-        String manifestFileSuffix = "-m0.avro";
-        // Iceberg seems to read manifest files twice during Streams.stream(combinedScanIterable) call in IcebergSplitSource
-        assertEquals(counts.forPathContaining(manifestFileSuffix).forOperation(INPUT_FILE_NEW_STREAM).sum(), 2);
-        // getLength is cached, so only assert number of different InputFile instances for a file
-        assertEquals(counts.forPathContaining(manifestFileSuffix).forOperation(INPUT_FILE_GET_LENGTH).get().size(), 1);
-        assertEquals(counts.forPathContaining(manifestFileSuffix).forOperation(INPUT_FILE_EXISTS).sum(), 0);
+        assertFileSystemAccesses("CREATE TABLE test_create (id VARCHAR, age INT)",
+                ImmutableMultiset.builder()
+                        .add(new FileOperation(SNAPSHOT, OUTPUT_FILE_LOCATION))
+                        .add(new FileOperation(METADATA, OUTPUT_FILE_LOCATION))
+                        .add(new FileOperation(METADATA, OUTPUT_FILE_CREATE))
+                        .add(new FileOperation(SNAPSHOT, OUTPUT_FILE_CREATE_OR_OVERWRITE))
+                        .add(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH))
+                        .add(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM))
+                        .build());
     }
 
-    private String runAndGetId(String query)
+    @Test
+    public void testCreateTableAsSelect()
+            throws Exception
+    {
+        assertFileSystemAccesses("CREATE TABLE test_create_as_select AS SELECT 1 col0",
+                ImmutableMultiset.builder()
+                        .add(new FileOperation(MANIFEST, OUTPUT_FILE_LOCATION))
+                        .add(new FileOperation(METADATA, OUTPUT_FILE_CREATE))
+                        .add(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH))
+                        .add(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(MANIFEST, OUTPUT_FILE_CREATE_OR_OVERWRITE))
```

Comment:
I tried to use  `addCopies` whenever it was feasible(the lines were identical, here there are not). Regarding using `addCopies(...,1)` instead of  `add` I will do it.